core/src/main/java/com/microsoft/applicationinsights/internal/config/connection/ConnectionString.java

```diff
@@ -35,7 +43,35 @@ public static void parseInto(String connectionString, TelemetryConfiguration tar
             throw new InvalidConnectionStringException("Could not parse connection string.");
         }
 
-        mapToConnectionConfiguration(kvps, targetConfig);
+        return kvps;
+    }
+
+    private static void mapStatsbeatConnectionStringToConfig(Map<String, String> kvps, TelemetryConfiguration config) throws InvalidConnectionStringException {
+        // get ikey
+        String instrumentationKey = kvps.get(Keywords.INSTRUMENTATION_KEY);
+        if (Strings.isNullOrEmpty(instrumentationKey)) {
+            throw new InvalidConnectionStringException("Missing '"+Keywords.INSTRUMENTATION_KEY+"'");
+        }
+        if (!Strings.isNullOrEmpty(config.getInstrumentationKey())) {
+            logger.warn("Connection string is overriding previously configured instrumentation key.");
+        }
+
+        config.setStatsbeatInstrumentationKey(instrumentationKey);
+
+        String suffix = kvps.get(Keywords.ENDPOINT_SUFFIX);
+        if (!Strings.isNullOrEmpty(suffix)) {
+            try {
+                config.getEndpointProvider().setStatsbeatEndpoint(constructSecureEndpoint(EndpointPrefixes.INGESTION_ENDPOINT_PREFIX, suffix));
+            } catch (URISyntaxException e) {
+                throw new InvalidConnectionStringException(Keywords.ENDPOINT_SUFFIX + " is invalid: " + suffix, e);
+            }
+        }
+
+        // set explicit endpoints for Statsbeat
+        String statsbeatEndpoint = kvps.get(Keywords.INGESTION_ENDPOINT);
+        if (!Strings.isNullOrEmpty(statsbeatEndpoint)) {
+            config.getEndpointProvider().setStatsbeatEndpoint(toUriOrThrow(statsbeatEndpoint, Keywords.INGESTION_ENDPOINT));
+        }
```

Comment:
if we don't expect users to change the statsbeat endpoint, we could configure instrumentationKey/ingestionEndpoint separately, and avoid parsing here